public/src/Helpers.ts

```diff
@@ -372,3 +312,30 @@ export const sortEnrollments = (enrollments: Enrollment[], sortBy: EnrollmentSor
         return enrollmentCompare(a, b, sortBy, descending)
     })
 }
+
+export class SubmissionsForCourse {
+    userSubmissions: { [key: string]: Submissions } = {}
```

Comment:
These could use the `Map` type with the `bigint` IDs as key instead of doing the conversion to string, or did I miss something?

Reply:
Ideally I would like to use `Map` for this. I'm having issues with Overmind not tracking mutations in the map when using `Map.set(x, y)`. I can get it to work if I clone the entire map, mutate the cloned map, then reassign the map in state with the mutated one. This seems to me to be terribly inefficient though.